### ISSUE TITLE
build/install fails from AUR/yay (setuptools-scm can't detect version from source tarball)

### ISSUE THREAD

**Issue by 0-andy-0** [insanum/gcalcli#793]
I have to admit I have no idea what to do with the following error when updating (yay gcalcli on manjaro)
```  File "/usr/lib/python3.12/site-packages/setuptools/dist.py", line 307, in __init__
    _Distribution.__init__(self, dist_attrs)
  File "/usr/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 284, in __init__
    self.finalize_options()
  File "/usr/lib/python3.12/site-packages/setuptools/dist.py", line 658, in finalize_options
    ep(self)
  File "/usr/lib/python3.12/site-packages/setuptools_scm/_integration/setuptools.py", line 123, in infer_version
    _assign_version(dist, config)
  File "/usr/lib/python3.12/site-packages/setuptools_scm/_integration/setuptools.py", line 58, in _assign_version
_version_missing(config)
  File "/usr/lib/python3.12/site-packages/setuptools_scm/_get_version_impl.py", line 117, in _version_missing
    raise LookupError(
LookupError: setuptools-scm was unable to detect version for .cache/yay/gcalcli/src/gcalcli-4.5.0.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
==> ERROR: A failure occurred in build().
    Aborting...
```

**Comment by dbarnett** [comment]
Manjaro builds from aur.archlinux.org/packages/gcalcli, correct? It's designed to get the version number from git labels, but I guess this is using a source snapshot that's missing .git dir.

Might be possible to force the version number somehow, but obviously workarounds shouldn't be needed to just yay update.

**Comment by dbarnett** [comment]
Commented on https://aur.archlinux.org/packages/gcalcli. I think it needs `SETUPTOOLS_SCM_PRETEND_VERSION_FOR_GCALCLI=$pkgver` in the AUR package to be able to build from a source tarball like that. Not sure if that's ideal or there's a better way for the AUR package to fetch the sources without losing attribution, but I did find another package https://aur.archlinux.org/packages/python-cmcrameri doing the same.

I'm not sure you can or should override that directly in your yay command but if you like you could try it in the meantime.

**Comment by dbarnett** [comment]
(I didn't see `SETUPTOOLS_SCM_PRETEND_VERSION_FOR_GCALCLI` added in the PKGBUILD yet so I still want to keep this open until we can verify it's fixed for pacman/yay without any workarounds.)

**Comment by dbarnett** [comment]
Should be fixed in the 4.5.0-3 update I just pushed!

**Comment by dbarnett** [comment]
FYI also just verified the fix takes care of yay (probably obvious, but I wasn't familiar with yay or manjaro before this)...
